### DISCUSSION TITLE
Stenographer\PCAP tuning?

### DISCUSSION THREAD

**Discussion by lonewaffle** [Security-Onion-Solutions/securityonion#4422] [Unsupported Versions] [answered]
Even with 6TB of storage, I'm only able to retain about 19 hours of PCAPs. Are there any options to tune stenographer to trim PCAPs for longer retention? Using 2.3.51.

Thanks!

**Comment by dougburks** [comment, marked as answer]
You can use a BPF to filter out traffic that you don't want to write to disk:
https://docs.securityonion.net/en/2.3/bpf.html

**Reply by lonewaffle** [reply]
Is it possible to have it only save PCAPs that are associated with alerts? I'm struggling to think of a way to use BPFs that are dynamic enough to capture PCAPs for situations that I'm interested in without capturing mostly junk that I don't.

**Reply by dougburks** [reply]
As shown in the docs, many folks apply a BPF to stenographer to filter out port 443 since the vast majority of port 443 traffic is encrypted and thus limited value from a packet capture perspective:
https://docs.securityonion.net/en/2.3/bpf.html#simple-example

**Comment by thegreekman** [comment]
> 
> 
> Even with 6TB of storage, I'm only able to retain about 19 hours of PCAPs. Are there any options to tune stenographer to trim PCAPs for longer retention? Using 2.3.51.
> 
> Thanks!

Did you ever figure a way to retain pcaps for longer? I as well can only retain about 24 hours of pcap with 5TB of storage

**Comment by lonewaffle** [comment]
I wish. I generally get between 12 and 19 hours of PCAP retention. Honestly, I wish Security Onion would hold onto PCAPs that were associated with alerts for much longer. I can’t always conduct an investigations in such a short amount of time with my other duties and convincing my supervisors to purchase more storage likely won’t fly on this one. Oh well, it’s still better than nothing.

-Josh

**Reply by dougburks** [reply]
@lonewaffle First, please keep in mind that disk space is incredibly cheap compared to the cost of a security incident. So you might consider putting it in business terms and submitting a cost benefit analysis to your supervisors showing that the purchase of additional storage would pay for itself after working a certain number of incidents.

Second, please note that sophisticated adversaries are going to attempt to evade your IDS alerts. Therefore, you want to ensure that you have sufficient storage to allow pcap retrieval even for flows that did not generate IDS alerts. For example, in our recent video (https://www.youtube.com/watch?v=5DxX2KAId7w) the most interesting streams had no IDS alerts whatsoever and if you didn't have the pcap you wouldn't see the attacker pillaging the victim's filesystem.

Finally, if you decide that you really need to store pcaps for alerts for a longer period of time, perhaps you could create a cron job to run periodically and query elasticsearch for alerts, iterate over the list, and then use so-pcap-export to export the pcap for long term archiving:
https://docs.securityonion.net/en/2.3/stenographer.html#command-line
